src/ExpressionEvaluator/Core/Source/ExpressionCompiler/DynamicFlagsCustomTypeInfo.cs

```diff
@@ -13,25 +12,55 @@ internal struct DynamicFlagsCustomTypeInfo
         /// <remarks>Internal for testing.</remarks>
         internal static readonly Guid PayloadTypeId = new Guid("826D6EC1-DC4B-46AF-BE05-CD3F1A1FD4AC");
 
-        private readonly BitArray _bits;
+        private readonly ReadOnlyCollection<byte> _bytes;
 
-        public DynamicFlagsCustomTypeInfo(BitArray bits)
+        private DynamicFlagsCustomTypeInfo(ReadOnlyCollection<byte> bytes)
+        {
+            _bytes = bytes;
+        }
+
+        public DynamicFlagsCustomTypeInfo(Guid payloadTypeId, ReadOnlyCollection<byte> payload)
+            : this(payloadTypeId == PayloadTypeId ? payload : null)
         {
-            _bits = bits;
         }
 
         public DynamicFlagsCustomTypeInfo(DkmClrCustomTypeInfo typeInfo)
+            : this(typeInfo != null && typeInfo.PayloadTypeId == PayloadTypeId ? typeInfo.Payload : null)
+        {
+        }
+
+        internal DynamicFlagsCustomTypeInfo(bool[] dynamicFlags)
```

Comment:
I like your suggestion, but I managed to break everything while trying to implement it (whereas the current version is already tested).  I'll submit a separate PR for this tomorrow.